### ISSUE TITLE
[rllib] DQfD Implementation

### ISSUE THREAD

**Issue by cedros23** [ray-project/ray#9167]
### What is your question?

Hello, 

I am trying to implement deep Q learning from demonstrations (DQfD) like algorithms. After the initial paper (https://arxiv.org/abs/1704.03732), Deepmind extended the algorithm to work with APE-X (https://arxiv.org/abs/1805.11593).

I have my own custom gym environment with the custom neural net model that works well with APE-X DQN. As a next stage, I want to boost training by expert demonstration data. I've checked the offline dataset part and created json file by expert demonstrations. 
As a next step, I integrated custom loss function including imitation loss with my custom model and custom env. Finally, I want to use a mixture of expert demonstrations and env simulation as in the offline dataset examples (https://docs.ray.io/en/master/rllib-offline.html). However, in the examples exploration is turned off and experience is mixed between sampler and expert demonstration which makes me confused since I still need my agent to explore the environment and go beyond expert demonstrations eventually as papers suggest.

As far as I understand, the sampler is only using current policy to generate new experience to update replay buffer. Apologies for the long explanations, 
long story short, how can I combine custom loss function (imitation loss, l2 loss etc.) as well as mixture of expert demonstrations and standard exploration?

Thanks & Regards,

I use master branch, python 3.6, torch 1.5

**Comment by Ajay-2007** [comment]
Hi @cedros23 , I have to same question but I have different custom env, I am also looking for workaround for this issue, If you get it to work please comment here.

Thank You,
Ajay
